candle-examples/examples/reinforcement-learning/ddpg.rs

```diff
@@ -0,0 +1,461 @@
+        varmap.set_one(
+            format!("{target_prefix}-fc{i}.weight"),
+            ((tau * network_w)? + ((1.0 - tau) * target_w)?)?,
+        )?;
+
+        let target_b = vb.get(out_dim, &format!("{target_prefix}-fc{i}.bias"))?;
+        let network_b = vb.get(out_dim, &format!("{network_prefix}-fc{i}.bias"))?;
+        varmap.set_one(
+            format!("{target_prefix}-fc{i}.bias"),
+            ((tau * network_b)? + ((1.0 - tau) * target_b)?)?,
+        )?;
+    }
+    Ok(())
+}
+
+struct Actor<'a> {
+    varmap: VarMap,
+    vb: VarBuilder<'a>,
+    network: Sequential,
+    target_network: Sequential,
+    size_state: usize,
+    size_action: usize,
+    dims: Vec<(usize, usize)>,
+}
+impl Actor<'_> {
+    fn new(device: &Device, dtype: DType, size_state: usize, size_action: usize) -> Result<Self> {
+        let varmap = VarMap::new();
+        let vb = VarBuilder::from_varmap(&varmap, dtype, device);
+
+        let dims: Vec<(usize, usize)> = vec![(size_state, 400), (400, 300), (300, size_action)];
+
+        let make_network = |prefix: &str| {
+            Ok::<Sequential, Error>(
+                seq()
+                    .add(linear(
+                        dims[0].0,
+                        dims[0].1,
+                        vb.pp(format!("{prefix}-fc0")),
+                    )?)
+                    .add(Activation::Relu)
+                    .add(linear(
+                        dims[1].0,
+                        dims[1].1,
+                        vb.pp(format!("{prefix}-fc1")),
+                    )?)
+                    .add(Activation::Relu)
+                    .add(linear(
+                        dims[2].0,
+                        dims[2].1,
+                        vb.pp(format!("{prefix}-fc2")),
+                    )?)
+                    .add(func(|xs| xs.tanh())),
+            )
+        };
+
+        let network = make_network("actor")?;
+        let target_network = make_network("target-actor")?;
+
+        // this sets the two networks to be equal to each other using TAU = 1.0
+        track(
+            varmap.clone(),
+            vb.clone(),
+            "target-actor",
+            "actor",
+            &dims,
+            1.0,
+        );
+
+        Ok(Self {
+            varmap,
+            vb,
+            network,
+            target_network,
+            size_state,
+            size_action,
+            dims,
+        })
+    }
+
+    fn forward(&self, state: &Tensor) -> Result<Tensor> {
+        self.network.forward(state)
+    }
+
+    fn target_forward(&self, state: &Tensor) -> Result<Tensor> {
+        self.target_network.forward(state)
+    }
+
+    fn track(&self, tau: f64) -> Result<()> {
+        track(
+            self.varmap.clone(),
+            self.vb.clone(),
+            "target-actor",
+            "actor",
+            &self.dims,
+            tau,
+        )
+    }
+}
+
+struct Critic<'a> {
+    varmap: VarMap,
+    vb: VarBuilder<'a>,
```

Comment:
Why is there a lifetime annotation here? I thought we had removed it from `VarBuilder` a long time ago.

Reply:
The compiler is telling me I need them:

```
error[E0106]: missing lifetime specifier
   --> candle-examples/examples/reinforcement-learning/ddpg.rs:255:9
    |
255 |     vb: VarBuilder,
    |         ^^^^^^^^^^ expected named lifetime parameter
    |
help: consider introducing a named lifetime parameter
    |
253 ~ struct Critic<'a> {
254 |     varmap: VarMap,
255 ~     vb: VarBuilder<'a>,
    |
```

Maybe because I'm keeping the `VarBuilder` in the struct?

Reply:
That's odd, you're using the latest github tip? Seems a bit like a relicate from a very old version.

Reply:
I forked a fresh version of candle after my messed up pull request and as you said transplanted my changes into it, so I guess so? The latest commit that is not mine is #1205.

Reply:
Ah ok, indeed it seems that I haven't cleaned the `VarBuilder` code as much as what I thought. I'll do that soonish but that seems good.